apps/namadillo/src/utils/assets.ts

```diff
@@ -0,0 +1,34 @@
+import { Asset } from "@chain-registry/types";
+import { IbcToken, NativeToken } from "@namada/indexer-client";
+import { shortenAddress } from "@namada/utils";
+
+export const unknownAsset = (denom: string): Asset => ({
+  denom_units: [{ denom, exponent: 0 }],
+  base: denom,
+  name: denom,
+  display: denom,
+  symbol: shortenAddress(denom, 4, 4),
+});
+
+export const findAssetByToken = (
```

Comment:
Can we keep the TODO comment of the original function?

Reply:
Ok! I'll update to refer to the `tnamAddressToDenomTrace` that is the new function. Maybe also exploring resolving this at all :)

Reply:
it'll be a bigger refactoring, so I left the comment 😁